packages/tools/accessibility/src/accessibilityTreeComponent.tsx

```diff
@@ -0,0 +1,242 @@
+            }
+
+            // If the node has GUI, add observer to the controls
+            if (this._isGUI(node)) {
+                const curMesh = node as AbstractMesh;
+                const adt = curMesh.material?.getActiveTextures()[0] as AdvancedDynamicTexture;
+                const guiRoot = adt.getChildren();
+                guiRoot.forEach((control) => addGUIObservers(control));
+            }
+        };
+
+        // observe add node and deal with new nodes
+        if (!this._observersMap.has(scene.onNewMeshAddedObservable)) {
+            this._observersMap.set(
+                scene.onNewMeshAddedObservable,
+                scene.onNewMeshAddedObservable.add((newNode) => {
+                    updateA11yTree();
+                    addNodeObservers(newNode);
+                })
+            );
+        }
+        if (!this._observersMap.has(scene.onNewTransformNodeAddedObservable)) {
+            this._observersMap.set(
+                scene.onNewTransformNodeAddedObservable,
+                scene.onNewTransformNodeAddedObservable.add((newNode) => {
+                    updateA11yTree();
+                    addNodeObservers(newNode);
+                })
+            );
+        }
+        // observe remove node
+        if (!this._observersMap.has(scene.onMeshRemovedObservable)) {
+            this._observersMap.set(scene.onMeshRemovedObservable, scene.onMeshRemovedObservable.add(updateA11yTree));
+        }
+        if (!this._observersMap.has(scene.onTransformNodeRemovedObservable)) {
+            this._observersMap.set(scene.onTransformNodeRemovedObservable, scene.onTransformNodeRemovedObservable.add(updateA11yTree));
+        }
+
+        // observe node enabled changed
+        scene.getNodes().forEach((node) => {
+            addNodeObservers(node);
+        });
+    }
+
+    componentWillUnmount() {
+        console.log("componentWillUnmount");
+        this._observersMap.forEach((observer, observable) => {
+            observable.remove(observer);
+        });
+        this._observersMap.clear();
+    }
+
+    render() {
+        return (
+            <div className="accessibility-tree">
+                {this.state.a11yTreeItems.map((item) => {
+                    return (
+                        <AccessibilityTreeItemComponent
+                            a11yItem={item}
+                            level={1}
+                            key={item.entity.uniqueId !== undefined && item.entity.uniqueId !== null ? item.entity.uniqueId : item.entity.name}
+                        />
+                    );
+                })}
+            </div>
+        );
+    }
+
+    private _updateAccessibilityTreeItems(): AccessibilityItem[] {
+        // Nodes
+        const rootNodes = this.props.scene.rootNodes.slice(0);
+        for (const mesh of this.props.scene.meshes) {
+            // Adding nodes that are parented to a bone
+            if (mesh.parent && mesh.parent.getClassName() === "Bone") {

```

Comment:
Why this special case?

Reply:
Learned this from Inspector https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/inspector/src/components/sceneExplorer/sceneExplorerComponent.tsx#L434